assets/js/blocks/checkout/inner-blocks/checkout-collection-method/frontend.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { withFilteredAttributes } from '@woocommerce/shared-hocs';
+import { FormStep } from '@woocommerce/base-components/cart-checkout';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { CHECKOUT_STORE_KEY } from '@woocommerce/block-data';
+/**
+ * Internal dependencies
+ */
+import Block from './block';
+import attributes from './attributes';
+
+const FrontendBlock = ( {
+	title,
+	description,
+	showStepNumber,
+	children,
+	className,
+	showPrice,
+	showIcon,
+}: {
+	title: string;
+	description: string;
+	showStepNumber: boolean;
+	children: JSX.Element;
+	className?: string;
+	showPrice: boolean;
+	showIcon: boolean;
+} ) => {
+	const { checkoutIsProcessing, prefersCollection } = useSelect(
+		( select ) => {
+			const checkoutStore = select( CHECKOUT_STORE_KEY );
+			return {
+				checkoutIsProcessing: checkoutStore.isProcessing(),
+				prefersCollection: checkoutStore.prefersCollection(),
```

Comment:
Good for now. Depending on the outcome of the p2 we may want to move this to local state perhaps. Leave it for now.

Reply:
This is still local state.